tests/test_northup.py

```diff
@@ -1,73 +1,192 @@
+       updated_dataset[0].dq[340:360,340:360]=1
+       if save_mock_dataset:
+          outdir = os.path.join('./',dirname)
+          os.makedirs(outdir, exist_ok=True)
+          updated_dataset[0].save(filedir='./',filename=f'mock_offset{ang+north_angle}deg.fits')
+       updated_datalist.append(updated_dataset[0])
+
+    input_dataset = data.Dataset(updated_datalist)
     derot_dataset = northup(input_dataset)
+
     # save fits file
-    if save_derot_dataset is True:
-        derot_dataset.save(dirname,[mockname.split('.fits')[0]+'_derotated.fits'])
-
-    # read the original mock file and derotated file
-    im_input = input_dataset[0].data
-    roll_angle = input_dataset[0].pri_hdr['ROLL']
-    im_derot = derot_dataset[0].data
-    dq_input = input_dataset[0].dq
-    dq_derot = derot_dataset[0].dq
-    # the location for test, where the mock file has 1 in DQ
-    x_value1 = input_dataset[0].dq_hdr['X_1VAL']
-    y_value1 = input_dataset[0].dq_hdr['Y_1VAL']
-
-    # check if rotation works properly
-    assert(im_input[y_value1,x_value1] != im_derot[y_value1,x_value1])
-    assert(dq_input[y_value1,x_value1] != dq_derot[y_value1,x_value1])
-    
-    # check if the derotated DQ frame has no non-integer values (except NaN)
-    non_integer_mask = (~np.isnan(dq_derot)) & (dq_derot % 1 != 0)
-    non_integer_indices = np.argwhere(non_integer_mask)
-    assert(len(non_integer_indices) == 0)
-
-    # save comparison figure
-    if save_comp_figure is True:
-
-        fig, (ax0, ax1) = plt.subplots(nrows=1, ncols=2, sharex=True, figsize=(8,5))
-
-        ax0.set_title('Original Mock Data')
-        ax0.imshow(im_input,origin='lower')
-
-        ax1.set_title(f'Derotated Mock Data\n by {-roll_angle}deg counterclockwise')
-        ax1.imshow(im_derot,origin='lower')
-
-        outdir = os.path.join(os.path.dirname(__file__),dirname)
-        os.makedirs(outdir, exist_ok=True)
-        outfilename = 'compare_northup.png'
-        
-        plt.savefig(os.path.join(outdir,outfilename))
-
-        print(f"Comparison figure saved at {dirname+outfilename}")
-
-if __name__ == "__main__":
-    test_northup()
+    if save_derot_dataset:
+       outdir = os.path.join('./',dirname)
+       os.makedirs(outdir, exist_ok=True)
+       for derot_data in derot_dataset:
+          derot_data.save(filedir='./',filename=f'derot_offset{ang+north_angle}deg.fits')
+
+    for i, (input_data,derot_data) in enumerate(zip(input_dataset,derot_dataset)):
+        # read the original mock file and derotated file
+        im_input = input_data.data
+        im_derot = derot_data.data
+        dq_input = input_data.dq
+        dq_derot = derot_data.dq
+
+        # the location for test, where the mock file has 1 in DQ
+        x_value1 = input_dataset[0].ext_hdr['X_1VAL']
+        y_value1 = input_dataset[0].ext_hdr['Y_1VAL']
+
+        # check if rotation works properly
+        assert(im_input[y_value1,x_value1] != im_derot[y_value1,x_value1])
```

Comment:
It would be great to also programmatically check that the bright pixels have been rotated to the correct position (in addition to the bright pixels no longer being in the same place). We're going to be normally running these tests in automated mode, so we don't have the time to make the figure and visually inspect it. You could be able to prediction based on the rotation angle + center where the bright pixels will land up.

You can either do it in this PR, or you can make a new issue for this as an enhancement, and we pull this PR in.